### PR TITLE
Feature/normalized user stats

### PR DESCRIPTION
**Pull request by gecBurton**
After discussion with 10DS we have agreed that we need deprecate the existing `user-statistics` endpoint because:
1. it returns a record per (anonymised) user, this is more information than is actually used and puts a lot of strain on our infrastructure
2. we need to capture the 1-59 minute learnings

To this end we have a new endpoint `normalized-user-statistics`, this has the following improvements:
1. we capture the 1-59 minute data
2. the response data is normalized, 10DS has been supplied with some client side code to transform it back to the format they expect
3. the data is pre-aggregated, this reduces the amount of data sent and speeds up the response time by more than 10x
4. the query uses the ORM

This has been achieved by:
1. putting an index on the group by fields
2. using django CTE to help with computing the `bucketed_hours` 
3. paginating the endpoint so that even if the number of users gets large the query will only be handling 1,000 records at a time

This PR does not include removing the original endpoint, this may be done later.

### CODE REVIEW

File: pytest_tests/conftest.py [hunk 1]
```diff
@@ -100,6 +103,7 @@ def eric(create_user):
     return create_user(
         email="eric@co.gov.uk",
         date_joined="2000-01-02",
+        times_to_complete=[5, 10],
```
Comment (gecBurton) [inline pytest_tests/conftest.py:106]: i want to catch the `(0,1)` edge case

File: pytest_tests/conftest.py [hunk 2]
```diff
@@ -90,8 +92,9 @@ def daisy(create_user):
         email="daisy@co.gov.uk",
         date_joined="2000-01-02",
         grade="GRADE6",
+        times_to_complete=[30, 60],
         has_completed_pre_survey=True,
-        has_completed_post_survey=False,
+        has_completed_post_survey=True,
```
Comment (gecBurton) [inline pytest_tests/conftest.py:97]: i want to check that users with the same number of bucketed hours are counted correctly, daisy's `has_completed_post_survey` now matches with chris's

File: pytest_tests/conftest.py [hunk 3]
```diff
@@ -55,6 +55,7 @@ def alice(create_user):
     return create_user(
         email="alice@co.gov.uk",
         date_joined="2000-01-01",
+        grade="GRADE7",
```
Comment (gecBurton) [inline pytest_tests/conftest.py:58]: this the default argument, so no change, I have added it explicitly for clarity

File: one_big_thing/learning/models.py [hunk 4]
```diff
@@ -75,6 +84,13 @@ def determine_competency_level(self):
             level = determine_competency_levels(competency_answers)
         return level
 
+    class Meta:
+        indexes = [
+            models.Index(
+                fields=("department", "grade", "profession", "has_completed_pre_survey", "has_completed_post_survey")
+            )
+        ]
```
Comment (gecBurton) [inline one_big_thing/learning/models.py:92]: im adding the index to speed up the group by

File: one_big_thing/learning/api_views.py [hunk 5]
```diff
@@ -171,3 +186,58 @@ def get_learning_breakdown_data():
...
+        "has_completed_post_survey",
+        "bucketed_hours",
+    ]
+
+    results = (
+        cte.queryset()
+        .with_cte(cte)
+        .values(*group_and_order_by)
+        .annotate(user_count=Count("id"))
+        .order_by(*group_and_order_by)
```
Comment (gecBurton) [inline one_big_thing/learning/api_views.py:225]: the `order_by` is important for pagination

**Review by nmenezes0 — APPROVED** [review]
This is so much more understandable! Looks good.

**Pull request merged**